File: src/main/java/com/koddy/server/coffeechat/domain/model/link/MeetingLinkProvider.java
package com.koddy.server.coffeechat.domain.model.link;

import com.koddy.server.coffeechat.exception.CoffeeChatException;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

import java.util.Arrays;

import static com.koddy.server.coffeechat.exception.CoffeeChatExceptionCode.INVALID_MEETING_LINK_PROVIDER;

@Getter
@RequiredArgsConstructor
public enum MeetingLinkProvider {
    ZOOM("zoom"),
    GOOGLE("google"),
    ;

    private final String provider;

    public static MeetingLinkProvider from(final String provider) {
        return Arrays.stream(values())
                .filter(meetingLinkProvider -> meetingLinkProvider.provider.equals(provider))
                .findFirst()
                .orElseThrow(() -> new CoffeeChatException(INVALID_MEETING_LINK_PROVIDER));
    }
}

File: src/main/kotlin/com/koddy/server/coffeechat/domain/model/link/MeetingLinkProvider.kt
package com.koddy.server.coffeechat.domain.model.link

import com.koddy.server.coffeechat.exception.CoffeeChatException
import com.koddy.server.coffeechat.exception.CoffeeChatExceptionCode.INVALID_MEETING_LINK_PROVIDER

enum class MeetingLinkProvider(
    val provider: String,
) {
    ZOOM("zoom"),
    GOOGLE("google"),
    ;

    companion object {
        fun from(provider: String): MeetingLinkProvider {
            return entries.firstOrNull { it.provider == provider }
                ?: throw CoffeeChatException(INVALID_MEETING_LINK_PROVIDER)
        }
    }
}
